extensions/amp-3d-gltf/0.1/amp-3d-gltf.js

```diff
@@ -151,17 +151,20 @@ export class Amp3dGltf extends AMP.BaseElement {
       return Promise.resolve();
     }
 
-    const iframe = getIframe(this.win, this.element, TYPE, this.context_);
-    iframe.title = this.element.title || 'GLTF 3D model';
-    applyFillContent(iframe, true);
-    this.iframe_ = iframe;
-    this.unlistenMessage_ = devAssert(this.listenGltfViewerMessages_());
+    return getIframe(this.win, this.element, TYPE, this.context_).then(
```

Comment:
Since the main change to component code is that `getIframe` is now async, I'd like it if the diff remained stable in order to maintain the commit history.

Please update this PR so that changes follow this pattern.

Where we would create the iframe like this:

```js
  layoutCallback() {
    const iframe = getIframe(...);

    // (do things with iframe)

    return this.loadPromise(iframe);
  }
```

Let's split the second part into a separate method named something like `loadIframe_(iframe)`. This way, we prevent this extra indentation and the diff history remains as-is.

```js
  layoutCallback() {
    return getIframe(...).then((iframe) => this.loadIframe_(iframe));
  }

  loadIframe_(iframe) {
    // (do things with iframe)

    return this.loadPromise(iframe);
  }
```

Reply:
Done. Nice suggestion!